bus/bus.go

```diff
@@ -1353,5 +1378,66 @@ func (b *bus) Handler() http.Handler {
 
 // Shutdown shuts down the bus.
 func (b *bus) Shutdown(ctx context.Context) error {
+	close(b.shutdown)
+	b.shutdownWG.Wait()
+
 	return b.eas.SaveAccounts(ctx, b.accounts.ToPersist())
 }
+
+func (b *bus) interactionRecordingLoop() {
+	for {
+		select {
+		case <-b.shutdown:
+			return
```

Comment:
it's not that important but I think there's a chance you are potentially missing interactions here and need to try and record after the `for` loop?